File: rust/merge_two_sorted_lists/src/list_node.rs
// ListNode cannot be modified. It is defined by Leet Code.
#[derive(PartialEq, Eq, Clone, Debug)]
pub struct ListNode {
    pub val: i32,
    pub next: Option<Box<ListNode>>,
}

impl ListNode {
    #[allow(dead_code)]
    fn new(val: i32) -> Self {
        ListNode { val, next: None }
    }
}

File: cpp/merge_two_sorted_lists/list_node.cpp
#include "list_node.h"

ListNode::ListNode(int v) : val{ v }, next{} {}

ListNode::ListNode(int v, ListNode *n) : val{ v }, next{ n } {}

std::vector<int> ListNode::to_vec() const
{
    std::vector<int> v;
    for (auto *node{ this }; nullptr != node; node = node->next)
    {
        v.push_back(node->val);
    }

    return v;
}